pkg/scheduler/objects/application.go

```diff
@@ -663,8 +663,6 @@ func (sa *Application) IsReservedOnNode(nodeID string) bool {
 // If the reservation fails the function returns false, if the reservation is made it returns true.
 // If the node and ask combination was already reserved for the application this is a noop and returns true.
 func (sa *Application) Reserve(node *Node, ask *AllocationAsk) error {
-	sa.Lock()
-	defer sa.Unlock()
```

Comment:
We should create a reserveInternal that is unlocked and leave Reserve locked calling reserveInternal.
In the allocation track (part of the change call reserveInternal as you already have the lock.